lib/gis_robot_suite/descriptive_metadata_builder.rb

```diff
@@ -96,10 +95,11 @@ def extract_publishers(publisher_nodes)
     end
 
     def extract_pubdate(node)
-      pub_date = node.xpath('../../../../gmd:date/gmd:CI_Date/gmd:date/gco:Date', NS)
-      raise "Publication date is missing for #{bare_druid}." unless pub_date.any?
+      pub_date_node = node.xpath('gmd:date/gmd:CI_Date/gmd:dateType/gmd:CI_DateTypeCode[@codeListValue="publication"]', NS)
```

Comment:
Get the date that has a type of publication, and don't accidentally collect all dates.